mtp_api/apps/transaction/permissions.py

```diff
@@ -12,7 +14,12 @@ def has_permission(self, request, view):
 class IsOwnPrison(BasePermission):
 
     def has_permission(self, request, view):
-        return request.user.prisonusermapping.prisons.filter(pk=view.kwargs.get('prison_id')).exists()
+        try:
+            prison_user_mapping = request.user.prisonusermapping
+        except PrisonUserMapping.DoesNotExist:
+            return False
```

Comment:
Feels a bit clunky to use a try/except here; is there a way we could check instead of waiting for an exception?